samcli/local/apigw/local_apigw_service.py

```diff
@@ -526,6 +526,15 @@ def _parse_v2_payload_format_lambda_output(lambda_output: str, binary_types, fla
         status_code = json_output.get("statusCode") or 200
         headers = Headers(json_output.get("headers") or {})
 
+        # cookies is a new field in payload format version 2.0 (a list)
+        # https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html
+        # we need to move cookies to Set-Cookie headers.
+        if isinstance(json_output.get("cookies"), list):
+            # add all cookies to Set-Cookie header
+            headers["Set-Cookie"] = (
+                ";".join([cookie for cookie in [headers.get("Set-Cookie")] + json_output.get("cookies") if cookie])
+            ).strip()
```

Comment:
Please add a reference as to why this `Set-Cookie` format is correct.

In [RFC 6265](https://datatracker.ietf.org/doc/html/rfc6265#section-4.1.1) the syntax doesn't describe allowing concatenating cookies with `;`. Instead, cookies are on separate `Set-Cookie` values.

From [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie):

> To send multiple cookies, multiple `Set-Cookie` headers should be sent in the same response.

Reply:
Yep you're right. API Gateway only supported single `Set-Cookie` header some time ago and I read a outdated document. So I was trying to make it look like what API Gateway should do. I just found API Gateway can support multiple `Set-Cookie` headers now. Updated the code.